pipeline_components/aiplatform/aiplatform/custom_train_job/component.py

```diff
@@ -0,0 +1,147 @@
+        project_id (str): project id of the Google Cloud project.
+        model_display_name (str): Name of the new trained model version.
+        train_container_uri (str): Container URI for running train script.
+        serving_container_uri (str): Container URI for deploying the output model.
+        model (Model): Trained model output.
+        metrics (Metrics): Output metrics of trained model.
+        requirements (List[str]): Additional python dependencies for training script.
+        job_name (str): Name of training job.
+        hparams (Dict[str, str]): Hyperparameters (passed as a JSON serialised argument
+            to train script)
+        replica_count (int): Number of replicas (increase for distributed training).
+        machine_type (str): Machine type of compute.
+        accelerator_type (str): Accelerator type (change for GPU support).
+        accelerator_count (str): Accelerator count (increase for GPU cores).
+        parent_model (str): Resource URI of existing parent model (optional). If `None`,
+            a new model will be uploaded. Otherwise, a new model version for the parent
+            model will be uploaded.
+    Returns:
+        parent_model (str): Resource URI of the parent model (empty string if the
+            trained model is the first model version of its kind).
+    """
+    import json
+    import logging
+    import os.path
+    import time
+    import google.cloud.aiplatform as aip
+
+    logging.info(f"Using train script: {train_script_uri}")
+    script_path = "/gcs/" + train_script_uri[5:]
+    if not os.path.exists(script_path):
+        raise ValueError(
+            "Train script was not found. "
+            f"Check if the path is correct: {train_script_uri}"
+        )
+
+    job = aip.CustomTrainingJob(
+        project=project_id,
+        location=project_location,
+        staging_bucket=staging_bucket,
+        display_name=job_name if job_name else f"Custom job {int(time.time())}",
+        script_path=script_path,
+        container_uri=train_container_uri,
+        requirements=requirements,
+        model_serving_container_image_uri=serving_container_uri,
+    )
+    cmd_args = [
+        f"--train_data={train_data.path}",
+        f"--valid_data={valid_data.path}",
+        f"--test_data={test_data.path}",
+        f"--metrics={metrics.path}",
+        f"--hparams={json.dumps(hparams if hparams else {})}",
+    ]
+    uploaded_model = job.run(
+        model_display_name=model_display_name,
+        parent_model=parent_model,
+        is_default_version=(not parent_model),
+        args=cmd_args,
+        replica_count=replica_count,
+        machine_type=machine_type,
+        accelerator_type=accelerator_type,
+        accelerator_count=accelerator_count,
+    )
+
+    resource_name = f"{uploaded_model.resource_name}@{uploaded_model.version_id}"
+    model.metadata["resourceName"] = resource_name
+    model.metadata["containerSpec"] = {"imageUri": serving_container_uri}
```

Comment:
if this is just the image URI couldn't we just pass it as a string rather than as a dict? I'm saying this mainly because in the UI you can see string outputs quite nicely but not dicts

Reply:
It's required in this format to be compatible with the `google.VertexModel` type I think